src/status_im2/contexts/wallet/common/sheets/network_preferences/view.cljs

```diff
@@ -1,70 +1,104 @@
-                   :customization-color account-color}}])
 
-(defn- networks-list
-  [account-color]
-  [{:title        "Optimism"
-    :image        :icon-avatar
-    :image-props  {:icon (resources/get-network :optimism)
-                   :size :size-20}
-    :action       :selector
-    :action-props {:type                :checkbox
-                   :customization-color account-color}}
-   {:title        "Arbitrum"
-    :image        :icon-avatar
-    :image-props  {:icon (resources/get-network :arbitrum)
-                   :size :size-20}
-    :action       :selector
-    :action-props {:type                :checkbox
-                   :customization-color account-color}}])
+(defn- make-network-item
+  [{:keys [network-name] :as _network}
+   {:keys [title color on-change network-preferences state] :as _options}]
+  {:title        (or title (string/capitalize (name network-name)))
+   :image        :icon-avatar
+   :image-props  {:icon (resources/get-network network-name)
+                  :size :size-20}
+   :action       :selector
+   :action-props {:type                (if (= :default state)
+                                         :filled-checkbox
+                                         :checkbox)
+                  :customization-color color
+                  :checked?            (contains? network-preferences network-name)
+                  :on-change           on-change}})
 
 (defn- view-internal
-  [{:keys [on-save theme]}]
-  (let [{:keys [color address]} (rf/sub [:wallet/current-viewing-account])]
-    [:<>
-     [quo/drawer-top
-      {:title       (i18n/label :t/network-preferences)
-       :description (i18n/label :t/network-preferences-desc)}]
-     [quo/data-item
-      {:status          :default
-       :size            :default
-       :label           :none
-       :blur?           false
-       :card?           true
-       :title           (i18n/label :t/address)
-       :subtitle        address
-       :subtitle-type   :default
-       :container-style (merge style/data-item
-                               {:background-color (colors/theme-colors colors/neutral-2_5
-                                                                       colors/neutral-90
-                                                                       theme)})}]
-     [quo/category
-      {:list-type :settings
-       :data      (mainnet color)}]
-     [quo/category
-      {:list-type :settings
-       :label     (i18n/label :t/layer-2)
-       :data      (networks-list color)}]
-     [quo/bottom-actions
-      {:button-one-label (i18n/label :t/update)
-       :button-one-props {:disabled?           true
-                          :on-press            on-save
-                          :customization-color color}}]]))
+  []
+  (let [state                               (reagent/atom :default)
+        {:keys [color address
+                network-preferences-names]} (rf/sub [:wallet/current-viewing-account])
```

Comment:
rf/sub outside render function, is it intentional? so you won't have reactions on this subscription

Reply:
It's ok here. `network-preferences-names` used for initialise - `initial-network-preferences-names`